test/files/neg/t9847.check

```diff
@@ -4,9 +4,6 @@ t9847.scala:6: warning: discarded non-Unit value
 t9847.scala:6: warning: a pure expression does nothing in statement position
   def f(): Unit = 42
                   ^
-t9847.scala:7: warning: discarded non-Unit value
-  def g = (42: Unit)
-           ^
 t9847.scala:7: warning: a pure expression does nothing in statement position
```

Comment:
Perhaps the `pure expr` warning is exactly the same use case?

Reply:
I don't understand the (seemingly to me) subtle differences between the two.

Reply:
"statement position" means not at the end of the block, which is "result position." "Value discard" (or "unit insertion") moves an expression into statement position, so it's doubly warned. I think that, for consistency, it should also not warn on `"hi"` in the example below. It's a lame use case, and the compiler is trying to warn only in limited cases (non-side-effecty). But it's easy to use the same notational cue and not warn.

Reply:
I don't see the "hi" example.

Reply:
I re-commented, which is what `RC` is for.